src/Components/Components/src/RenderTree/Renderer.cs

```diff
@@ -658,6 +659,12 @@ internal void AddToRenderQueue(int componentId, RenderFragment renderFragment)
     {
         Dispatcher.AssertAccess();
 
+        if (_rendererIsStopped)
+        {
+            // Once we're stopped, we'll disregard further attempts to queue anything
+            return;
+        }
+
```

Comment:
I think we should do this inside `https://github.com/dotnet/aspnetcore/blob/main/src/Components/Components/src/RenderTree/Renderer.cs#L736` which can be overriden by `EndpointHtmlRenderer` and save ourselves having the public API `SignalRendererToFinishRendering`

Reply:
if we do as you propose, the queue will keep being edited during the processing (line 676 below). That was why I added a new public API, to prevent new Enqueue() when we already sent the `stop` signal.